.travis.yml

```diff
@@ -4,6 +4,7 @@ rust:
 
 os:
   - linux
+  - osx
```

Comment:
I am not sure if it is a good idea to also test the compile process on OS-X. I think it exceeds the scope of tests we should run for libtock-rs but rather tests that cross-compiling with cargo works.

Reply:
There has been at least one OSX-specific failure in the past (see linked comment), especially due to the custom linker scripts. But the extra overhead may not be worth it, as such OS-specific issues don't happen really often.

Reply:
I have to say I am opposed to testing the toolchain by running the whole CI for different host operating systems. As up to now we had no compatibility issues with MacOS so my solution would be to

 * continue to cleanly compile parts only for "none" architectures which are only meant only for "none" architectures (as in the example of syscalls)
 * close this PR.

What do you think?

Reply:
Note that the "none" architecture configuration has been quite controversial lately, see the discussion in https://github.com/tock/tock/pull/1393

Reply:
> As up to now we had no compatibility issues with MacOS

There was a compatibility issue in https://github.com/tock/tock/pull/1393#issuecomment-539723255, which motivates this PR.

That said, I'm fine with having Linux being "first class" supported, and OSX only supported on a best-effort basis. But it's up to @tock/core-wg to decide the level of support for each platform.

Reply:
For me, it's ok to have the osx toolchain running. It makes sense for the binaries since people already discovered incompatibilities in the linker scripts and it makes sense for the tests which are run in parallel and could require os-specific tooling like thread-locals in the future.

So would @torfmaster be okay with merging this?

Reply:
I'm okay with trying this. If it slows down the build too much we can reconsider removing the target.